src/automata.jl

```diff
@@ -102,50 +120,76 @@ constitute `AbstractAutomaton` interface:
  * `walk`: traveling through the automaton according to given path and initial state
 
 """
-abstract type AbstractAutomaton end
+abstract type AbstractAutomaton{T, N} end
 
 """
     Automaton{T} <: AbstractAutomaton
 Automaton as a vector of states together with alphabet.
 """
-struct Automaton{T} <: AbstractAutomaton
-    states::Vector{State{T}}
+struct Automaton{T, N} <: AbstractAutomaton{T, N}
+    states::Vector{State{T, N}}
     abt::Alphabet
+    uniquenoedge::State{T, N}
 end
 
-# setting the default type to UInt16
-Automaton(states::Vector{Union{State{T}, AbstractState{T}}}, abt::Alphabet) where {T<:Integer} = Automaton{UInt16}(states, abt)
-
-Automaton(abt::Alphabet) = Automaton([State(Word(Int[]), length(abt))], abt)
+# Default type is set to UInt16
+function Automaton(abt::Alphabet)
+    uniquenoedge = State{UInt16, length(abt)}()
+    Automaton{UInt16, length(abt)}([State(Word(), length(abt), uniquenoedge)], abt, uniquenoedge)
+end
 
 states(a::Automaton) = a.states
 initialstate(a::Automaton) = states(a)[1]
+noedge(a::Automaton) = a.uniquenoedge
+
+Base.push!(a::Automaton{T}, s::State{T}) where {T} = push!(a.states, s)
+Base.push!(a::Automaton{T}, name::AbstractWord{T}) where {T} = push!(states(a), State(name, length(a.abt), noedge(a)))
 
-function Base.push!(a::Automaton{T}, s::State{T}) where {T}
-    @assert (length(s.ined) == length(a)) && (length(s.outed) == length(a)) "Tables of in and out edges must have the same length as alphabet"
-    push!(a.states, s)
+"""
+    replace(k::NTuple{N, T}, val, idx::Integer) where {N, T}
+Returns a copy of `NTuple` `k` with value at index `idx` changed to `val`.
+"""
+function replace(k::NTuple{N, T}, val, idx::Integer) where {N, T}
+    @boundscheck 1 ≤ idx ≤ N || throw(BoundsError("Tried to access $(typeof(k)) at index $idx"))
```

Comment:
I actually didn't test what I wrote:) this is correct
```suggestion
    @boundscheck 1 ≤ idx ≤ N || throw(BoundsError(k, idx))
```